spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/CassandraRow.scala

```diff
@@ -86,18 +140,23 @@ object CassandraRow {
     * the newly created `CassandraRow`, but it is not used to fetch data from
     * the input `Row` in order to improve performance. Fetching column values by name is much
     * slower than fetching by index. */
-  def fromJavaDriverRow(row: Row, columnNames: Array[String]): CassandraRow = {
-    val data = new Array[Object](columnNames.length)
-    for (i <- columnNames.indices)
-      data(i) = GettableData.get(row, i)
-    new CassandraRow(columnNames, data)
+  def fromJavaDriverRow(row: Row, metaData: CassandraRowMetaData): CassandraRow = {
+    val data = new Array[Object](metaData.columnNames.length)
+    for (i <- metaData.columnNames.indices) {
+      data(i) = metaData.codecs match {
```

Comment:
Fixed, I also replace for() with mutable while() to reduce number of created objects